public/plugins/query_editors/query_header.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiButton, EuiButtonGroup } from '@elastic/eui';
+
+const QUERY_MODE_OPTIONS = [
+  { label: 'Builder', id: 'visual_builder' },
+  { label: 'Code', id: 'query_editor' },
+];
+
+export const QueryModeSwitcher = () => {
+  const [toggleCompressedIdSelected, setToggleCompressedIdSelected] = useState('query_editor');
```

Comment:
ideally
```typescript
const QUERY_MODE_OPTIONS = [
  { label: 'Builder', id: 'visual_builder' },
  { label: 'Code', id: 'query_editor' },
] as const satisfies EuiButtonGroupOptionProps[];

export const QueryModeSwitcher = () => {
  const [toggleCompressedIdSelected, setToggleCompressedIdSelected] =
    useState<(typeof QUERY_MODE_OPTIONS)[number]['id']>('query_editor');
```
might be too much and not sure if OSD's typescript version supports `satisfies`

Reply:
seems like not supported. but type EuiButtonGroupOptionProps[] makes sense